classes/01class/exercises/c01-aws01/raghunadhpokkalath/README.md

```diff
@@ -0,0 +1,441 @@
+                    "InterfaceType": "interface"
+                }
+            ],
+            "RootDeviceName": "/dev/xvda",
+            "RootDeviceType": "ebs",
+            "SecurityGroups": [
+                {
+                    "GroupName": "ec2-sg",
+                    "GroupId": "sg-0c43ee1f951df16a7"
+                }
+            ],
+            "SourceDestCheck": true,
+            "StateReason": {
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "361529347535",
+    "ReservationId": "r-0fa0e40ec0f1a8e22"
+}
+
+Get Public ip address
+> aws ec2 describe-instances --instance-ids i-0c98ac03e17a8323b --query 'Reservations[0].Instances[0].PublicIpAddress'
+3.25.132.226
+
+```
+
+- Commands to connect to the first EC2 instance:
+```
+> ssh -i "EC2KeyPair.pem" ec2-user@3.25.132.226
+
+The authenticity of host '3.25.132.226 (3.25.132.226)' can't be established.
+ECDSA key fingerprint is SHA256:mshfO7fCSeQ7HJtuVDoMi+a5ehKfI0CaqJasw8FtLrY.
+Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
+Warning: Permanently added '3.25.132.226' (ECDSA) to the list of known hosts.
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+Permissions 0644 for 'EC2KeyPair.pem' are too open.
+It is required that your private key files are NOT accessible by others.
+This private key will be ignored.
+Load key "EC2KeyPair.pem": bad permissions
+ec2-user@3.25.132.226: Permission denied (publickey).
+
+Change permission of Pem file
+chmod 400 EC2KeyPair.pem
+
+>ssh -i "EC2KeyPair.pem" ec2-user@3.25.132.226
+
+https://aws.amazon.com/amazon-linux-ami/2018.03-release-notes/
+5 package(s) needed for security, out of 7 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-47-59 ~]$ 
+
+```
+
+- Commands to create the second EC2 instance and any additional resource required:
+```
+Create new security group  to allow ssh only from same VPC 
+
+> aws ec2 create-security-group --group-name ec2-sg-private --description "EC2 Security Group Private connection"
+
+{
+    "GroupId": "sg-04a824ffed232066c"
+}
+Add CIDR to security group to allow ssh connection only from same VPC
+
+> aws ec2 authorize-security-group-ingress --group-id sg-04a824ffed232066c --protocol tcp --port 22 --cidr 172.31.0.0/16
```

Comment:
it's good that you are restricting the access to your local networking by CIDR, but in this case, will be best if you reference the security group from the other instance. Can you update that?

Reply:
Thanks, @caiocezart modified the code to use  other instance security group using --source-group  instead of referring CIDR